File: app/src/main/java/com/arjanvlek/oxygenupdater/Model/ServerMessage.java
package com.arjanvlek.oxygenupdater.Model;

import com.fasterxml.jackson.annotation.JsonProperty;

public class ServerMessage {
    private long id;
    private String englishMessage;
    private String dutchMessage;
    private Long deviceId;
    private Long updateMethodId;
    private ServerMessagePriority priority;

    public enum ServerMessagePriority {
        LOW, MEDIUM, HIGH
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getEnglishMessage() {
        return englishMessage;
    }

    @JsonProperty("english_message")
    public void setEnglishMessage(String englishMessage) {
        this.englishMessage = englishMessage;
    }

    public String getDutchMessage() {
        return dutchMessage;
    }

    @JsonProperty("dutch_message")
    public void setDutchMessage(String dutchMessage) {
        this.dutchMessage = dutchMessage;
    }

    public long getDeviceId() {
        return deviceId;
    }

    @JsonProperty("device_id")
    public void setDeviceId(Long deviceId) {
        this.deviceId = deviceId;
    }

    public long getUpdateMethodId() {
        return updateMethodId;
    }

    @JsonProperty("update_method_id")
    public void setUpdateMethodId(Long updateMethodId) {
        this.updateMethodId = updateMethodId;
    }

    public ServerMessagePriority getPriority() {
        return priority;
    }

    public void setPriority(ServerMessagePriority priority) {
        this.priority = priority;
    }

}

File: app/src/main/java/com/arjanvlek/oxygenupdater/models/ServerMessage.kt
package com.arjanvlek.oxygenupdater.models

import android.content.Context
import androidx.core.content.ContextCompat
import com.arjanvlek.oxygenupdater.R
import com.arjanvlek.oxygenupdater.internal.i18n.AppLocale

data class ServerMessage(
    var id: Long = 0,
    var englishMessage: String? = null,
    var dutchMessage: String? = null,
    var deviceId: Long? = null,
    var updateMethodId: Long? = null,
    var priority: ServerMessagePriority? = null
) : Banner {

    override fun getBannerText(context: Context?): String? {
        return if (AppLocale.get() == AppLocale.NL) dutchMessage else englishMessage
    }

    override fun getColor(context: Context?): Int {
        return when (priority) {
            ServerMessagePriority.LOW -> ContextCompat.getColor(context!!, R.color.colorPositive)
            ServerMessagePriority.MEDIUM -> ContextCompat.getColor(context!!, R.color.colorWarn)
            ServerMessagePriority.HIGH -> ContextCompat.getColor(context!!, R.color.colorPrimary)
            else -> ContextCompat.getColor(context!!, R.color.colorPositive)
        }
    }

    enum class ServerMessagePriority {
        LOW,
        MEDIUM,
        HIGH
    }
}
